components/tag-list/tag-list.js

```diff
@@ -228,19 +230,34 @@ class TagList extends LocalizeCoreElement(ArrowKeysMixin(LitElement)) {
 	}
 
 	_handleResize(entries) {
-		this._availableWidth = Math.ceil(entries[0].contentRect.width);
+		this._availableWidth = Math.floor(entries[0].contentRect.width);
 		if (this._availableWidth >= PAGE_SIZE.large) this._lines = PAGE_SIZE_LINES.large;
 		else if (this._availableWidth < PAGE_SIZE.large && this._availableWidth >= PAGE_SIZE.medium) this._lines = PAGE_SIZE_LINES.medium;
 		else this._lines = PAGE_SIZE_LINES.small;
+		if (!this._hasResized) {
+			this._hasResized = true;
+			this._handleSlotChange();
+		}
 		this._chomp();
 	}
 
 	_handleSlotChange() {
-		requestAnimationFrame(() => {
+		if (!this._hasResized) return;
+
+		requestAnimationFrame(async() => {
 			this._items = this._getTagListItems();
-			this._itemLayouts = this._getItemLayouts(this._items);
+			if (!this._items || this._items.length === 0) return;
+
+			const numItems = this._items.length;
+			const updateItems = new Promise((resolve) => {
+				this._items.forEach(async(item, index) => {
+					await item.updateComplete;
+					if (index === numItems - 1) resolve();
+				});
+			});
+			await updateItems;
 
-			if (this._items.length === 0) return;
+			this._itemLayouts = this._getItemLayouts(this._items);
```

Comment:
I added an add/remove button to the demo page to test that behaviour.  The chomping seems to not work quite right when adding items ... it seems like once the items overflow, the `_getItemLayouts` doesn't always return updated results.  The remove doesn't appear to work as expected either.

Reply:
Okay interesting - I'll dig more into this

Reply:
Ah so it is returning updated results but because some of them are hidden after the third addition (with the screen size I'm testing) they're getting filtered out (https://github.com/BrightspaceUI/core/blob/2a071093a783cc93dc0f17f2ef48d4aff3f7ac35/components/tag-list/tag-list.js#L212). I can probably check for the `data-is-chomped` and not filter them in that case even if hidden